### PR TITLE
issue #410 - index the display unit field even when a coded unit is present

### PR DESCRIPTION
**Pull request by lmsurpre**
see https://chat.fhir.org/#narrow/stream/179166-implementers/topic/Quantity.20search.20and.20display.20units/near/180734154

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java [hunk 1]
```diff
@@ -452,21 +453,31 @@ public boolean visit(java.lang.String elementName, int elementIndex, Quantity qu
         if (!QUANTITY.equals(searchParamType)) {
             throw invalidComboException(searchParamType, quantity);
         }
-        if (quantity != null && quantity.getValue() != null && quantity.getValue().getValue() != null &&
-                (quantity.getCode() != null || quantity.getUnit() != null)) {
-            Parameter p = new Parameter();
-            p.setName(searchParamCode);
-            p.setValueNumber(quantity.getValue().getValue());
+        if (quantity.getValue() != null && quantity.getValue().hasValue()) {
```
Comment (albertwang-ibm) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java]: so, don't have to do null check for quantity any more?

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchQuantityTest.java [hunk 2]
```diff
@@ -37,14 +37,10 @@ protected void setTenant() throws Exception {
...
-        // DSTU2 does not say if this is allowed or not, but we do not support it.
-        // In more recent versions, they clarified that it should work:  https://build.fhir.org/search.html#quantity
-//        assertSearchReturnsSavedResource("Quantity", "25");
-        
-        // I think this should return the resource but it currently doesn't.
         // https://gforge.hl7.org/gf/project/fhir/tracker/?action=TrackerItemEdit&tracker_item_id=19597
-//        assertSearchReturnsSavedResource("Quantity", "25||sec");
+        assertSearchReturnsSavedResource("Quantity", "25||sec");
         
         assertSearchDoesntReturnSavedResource("Quantity", "24.4999||s");
```
Comment (albertwang-ibm) [inline fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchQuantityTest.java]: according to the spec https://www.hl7.org/fhir/search.html,  GET [base]/Observation?value-quantity=5.4||mg  means "Search for all the observations with a value of 5.4(+/-0.05) mg where the unit - either the code (code) or the stated human unit (unit) are "mg",   do we do the +/- value range check also?

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java]: right, this makes it consistent with the other visit methods...they all assume a non-null argument and this should always be the case from FHIRPersistenceJDBCImpl due to the double-dispatch logic

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchQuantityTest.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchQuantityTest.java]: No, that is https://github.com/IBM/FHIR/issues/257

**Pull request merged**